docs/core-components-2.md

```diff
@@ -0,0 +1,83 @@
+The AEA framework consists of several core elements, some which are required to run an AEA and others which are optional.
+
+## The advanced elements AEAs use
+
+In <a href="../core-components-1/">Core Components - Part 1</a> we discussed the elements each AEA uses. We will now look at some of the advanced elements each AEA uses.
+
+### Decision Maker
+
+The `DecisionMaker` component manages global agent state updates proposed by the skills and processes the resulting ledger transactions.
+
+It is responsible for the AEA's crypto-economic security and goal management, and it contains the preference and ownership representation of the AEA.
+
+Skills communicate with the decision maker via `InternalMessages`. There exist two types of these: `TransactionMessage` and `StateUpdateMessage`.
+
+The `StateUpdateMessage` is used to initialize the decision maker with preferences and ownership states. It can also be used to update the ownership states in the decision maker if the settlement of transaction takes place off chain.
+
+The `TransactionMessage` is used by a skill to propose a transaction to the decision maker. The performative `TransactionMessage.Performative.PROPOSE_FOR_SETTLEMENT` is used by a skill to propose a transaction which the decision maker is supposed to settle on chain. The performative `TransactionMessage.Performative.PROPOSE_FOR_SIGNING` is used by the skill to propose a transaction which the decision maker is supposed to sign and which will be settled later.
+
+The decision maker processes messages and can accept or reject them.
+
+<div class="admonition note">
+  <p class="admonition-title">Note</p>
+  <p>For examples how to use these concepts have a look at the `tac_` skills. These functionalities are experimental and subject to change.
+</p>
+</div>
+
+### Wallet
+
+The wallet contains the private-public key pairs used by the AEA.
+
+### Identity
+
+The identity contains the AEAs addresses as well as its name.
```

Comment:
We could add other words such as:

> The identity is an abstraction that represents the identity of the agent in the Open Economic Framework, backed by public-key cryptography.

Though it might sound more special than it is, given the current implementation. Some words on Sovrin?

Optional: It is accessible by the skill code (i.e. business logic) through the agent context.

Reply:
We don't provide any integration with Sovrin atm.